library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java

```diff
@@ -561,7 +561,7 @@ private void doSomeWork() throws ExoPlaybackException, IOException {
     TraceUtil.beginSection("doSomeWork");
 
     updatePlaybackPositions();
-    long rendererPositionElapsedRealtimeUs = SystemClock.elapsedRealtime() * 1000;
+    long rendererPositionElapsedRealtimeUs = (SystemClock.elapsedRealtime() * 1000) + mediaClock.getPlaybackParameters().videoOffsetUs;
```

Comment:
This isn't really the right place to apply this offset. It should probably be applied somewhere under `DefaultAudioSink.getCurrentPositionUs`, if anywhere.

Reply:
@ojw28  oh i see, this is works too. you are right, this logic is more relevant to be applied into class DefaultAudioSink.
OK I will move this part into  `DefaultAudioSink.getCurrentPositionUs`.